pandas/_libs/tslibs/fields.pyx

```diff
@@ -254,7 +253,7 @@ def get_start_end_field(
         # month of year. Other offsets use month, startingMonth as ending
         # month of year.
 
-        if to_offset(freqstr).name[0:2]  in ["MS", "QS", "YS"]:
+        if freqstr[0:2]  in ["MS", "QS", "YS"]:
```

Comment:
we should also rename the variable name now if something else is being passed in (`freq_name`?)